server/protocols/jmap/src/main/java/org/apache/james/jmap/crypto/JwtTokenVerifier.java

```diff
@@ -0,0 +1,58 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.jmap.crypto;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.jsonwebtoken.*;
+
+import javax.inject.Inject;
+import java.util.Optional;
+
+public class JwtTokenVerifier {
+
+    private final PublicKeyProvider pubKeyProvider;
+
+    @Inject
+    @VisibleForTesting
+    JwtTokenVerifier(PublicKeyProvider pubKeyProvider) {
+        this.pubKeyProvider = pubKeyProvider;
+    }
+
+    public boolean verify(String token) {
+        return parseToken(token).isPresent();
+    }
+
+    public String extractLogin(String token) {
+        return parseToken(token)
+                .map(Jwt::getBody)
+                .map(Claims::getSubject)
+                .get();
+    }
+
+    private Optional<Jws<Claims>> parseToken(String token) {
+        try {
+            return Optional.ofNullable(Jwts
+                    .parser()
+                    .setSigningKey(pubKeyProvider.get())
+                    .parseClaimsJws(token));
+        } catch(ExpiredJwtException | UnsupportedJwtException | MalformedJwtException | SignatureException e ) {
```

Comment:
not convinced at all about exception catching in all cases, I think that the decision is up to the caller